framework/src/meshmodifiers/ParsedAddSideset.C

```diff
@@ -0,0 +1,114 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// MOOSE includes
+#include "ParsedAddSideset.h"
+#include "Conversion.h"
+#include "MooseMesh.h"
+
+// libmesh includes
+#include "libmesh/fparser_ad.hh"
+
+template<>
+InputParameters validParams<ParsedAddSideset>()
+{
+  InputParameters params = validParams<AddSideSetsBase>();
+  params += validParams<FunctionParserUtils>();
+  params.addRequiredParam<std::string>("combinatorial_geometry", "Function expression encoding a combinatorial geometry");
+  params.addRequiredParam<BoundaryName>("new_sideset_name", "The name of the new sideset");
+  params.addParam<std::vector<SubdomainID> >("included_subdomain_ids", "A set of subdomain ids whose sides will be included in the new sidesets");
+  params.addParam<Point>("normal", "If provided specifies the normal vector on sides that are added to the new ");
+  params.addParam<std::vector<std::string> >("constant_names", "Vector of constants used in the parsed function (use this for kB etc.)");
+  params.addParam<std::vector<std::string> >("constant_expressions", "Vector of values for the constants in constant_names (can be an FParser expression)");
+  params.addClassDescription("A MeshModifier that adds element's sides to a sideset if the centroid satisfies the combinatorial_geometry expression, (and optionally) "
+                             "if one of the side's elements is in included_subdomain_ids and if it features the correct normal.");
+  return params;
+}
+
+ParsedAddSideset::ParsedAddSideset(const InputParameters & parameters) :
+    AddSideSetsBase(parameters),
+    FunctionParserUtils(parameters),
+    _function(parameters.get<std::string>("combinatorial_geometry")),
+    _sideset_name(getParam<BoundaryName>("new_sideset_name")),
+    _check_subdomains(isParamValid("included_subdomain_ids")),
+    _check_normal(isParamValid("normal")),
```

Comment:
you could do the isParamSetByUser check here and give it a a default value of `Point()` for one less check below